packages/input-selection/src/ChangeAddress/StaticChangeAddressResolver.ts

```diff
@@ -0,0 +1,40 @@
+import { Cardano } from '@cardano-sdk/core';
+import { ChangeAddressResolver, Selection } from '../';
+import { GroupedAddress } from '@cardano-sdk/key-management';
+import { InvalidStateError } from '@cardano-sdk/util';
+import { Observable, firstValueFrom } from 'rxjs';
+
+/**
+ * Default change address resolver.
+ */
+export class StaticChangeAddressResolver implements ChangeAddressResolver {
+  readonly #addresses$: Observable<GroupedAddress[]>;
+
+  /**
+   * Initializes a new instance of the StaticChangeAddressResolver.
+   *
+   * @param addresses$ The observable with the known addresses.
+   */
+  constructor(addresses$: Observable<GroupedAddress[]>) {
```

Comment:
`input-selection` doesn't have a dependency on `rxjs` and it would be great to keep it that way
```suggestion
  constructor(getAddresses: () => Promise<GroupedAddress[]>) {
```